src/main/java/org/spongepowered/api/world/Location.java

```diff
@@ -666,24 +666,24 @@ public Entity createEntity(EntityType type) {
      * should be taken note that there can be many reasons for a failure.</p>
      *
      * @param entity The entity to spawn
-     * @return True if successful, false if not
+     * @return The entity, if it was spawned successfully
      * @see EntityUniverse#spawnEntity(Entity)
      */
-    public boolean spawnEntity(Entity entity) {
+    public Optional<Entity> spawnEntity(Entity entity) {
```

Comment:
I'm not sure I can allow a break for sugaring in this case.

Reply:
Fair enough. I will let it sizzle for a bit, but this one I am cool with reverting.

Reply:
Would the returned entity ever _not_ be the entity given in the parameter? If not, I don't see a reason for this

Reply:
It shouldn't be any other entity :P. This was kind of just for convenience, but yeah not really worth a breaking chance.

`world.spawnEntity(location.createEntity(EntityTypes.BOAT)).ifPresent(e -> e.addPassenger(player))` or something.

Reply:
It was more for consistency with the rest of the API / spawnEntities being a collection.

Reply:
This change would break a lot more plugins and overall it doesn't do much for the developer, so I will revert it.